src/main/java/frameworksdrivers/StudentDbGateway.java

```diff
@@ -67,24 +67,23 @@ public void getUser(LoginStudentDbRequestModel dbRequestModel) {
         }
     }
 
-    /**
-     * Method returns whether the username is unique meaning that it's not already present in the database.
-     *
-     * @param username is the username that is checked.
-     * @return boolean that indicates if username is unique.
-     */
-    @Override
-    public boolean isUnique(String username) {
-        return true;
-    }
-
     /**
      * Method saves the user info stored in "student" to the database.
      *
      * @param student model that stores what will be saved onto the database.
      */
     @Override
-    public void save(CreateStudentDsModel student) {
-        System.out.println("It saved!");
+    public void saveUser(CreateStudentDsModel student) {
+        try{
+        Map<String, Integer> courses = student.getCourseList();
+        List<String> courseNames = (List<String>) courses.keySet();
+        for(String course: courseNames){
+            courses.get(course);
+            PreparedStatement statement = connection.prepareStatement("INSERT INTO student (StudentID, Password," +
+                    " CourseID, CourseGrade) Values (" +student.getUsername() +", " + student.getPassword() +", "+
+                    course +", "+ courses.get(course)+")");
```

Comment:
I'm not sure if this is exactly going to execute the way we intend to. SQL databases are very particular with prepareStatements. I think in the prepared statement it should've been

"INSERT INTO student (StudentID, Password, CourseID, CourseGrade) VALUES (?, ?, ?, ?)"

And in the next line you would insert the correct values for the question mark. However I think this will work, and otherwise we'll make changes.